### PR TITLE
Bump MDI and add warning for removed and renamed icons

### PR DESCRIPTION
**Pull request by bramkragten**
None breaking for now, user has 2 versions to change removed/renamed icons.

Home Assistant Core PR: https://github.com/home-assistant/core/pull/37730

## Type of change

- [X] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by iantrich** [comment]
Really nice way of handling this 👍

File: src/components/ha-icon.ts [hunk 1]
```diff
@@ -21,6 +21,200 @@ import {
...
+  "xbox-controller-battery-unknown":
+    "microsoft-xbox-controller-battery-unknown",
+  "xbox-controller-menu": "microsoft-xbox-controller-menu",
+  "xbox-controller-off": "microsoft-xbox-controller-off",
+  "xbox-controller-view": "microsoft-xbox-controller-view",
+  yammer: "microsoft-yammer",
+  "youtube-creator-studio": "youtube-studio",
+};
+
+const mdiRemovedIcons = [
```
Comment (balloob) [inline src/components/ha-icon.ts:150]: Make this a Set `new Set([…])`

File: src/entrypoints/core.ts [hunk 2]
```diff
@@ -110,24 +109,3 @@ window.hassConnection.then(({ conn }) => {
     llWindow.llResProm = fetchResources(conn);
   }
 });
-
-window.addEventListener("error", (e) => {
```
Comment (balloob) [inline src/entrypoints/core.ts:114]: Why remove this?

Comment (balloob) [reply src/entrypoints/core.ts:114]: I see now that you moved this. We should not. 

This needs to be loaded as soon as possible or else we're going to miss events. 

If you prefer to move some logic to a later place, at least start listening for events and store them in a list for later processing.

**Review by balloob — APPROVED** [review]
Nice way of dealing with the upgrade! That will make a lot of people happy.

File: src/entrypoints/core.ts (hunk 2 above)
Comment (bramkragten) [reply src/entrypoints/core.ts:114]: It only did something when hass was available, I made sure it is still added before hass is initialized

Comment (balloob) [reply src/entrypoints/core.ts:114]: but what if `core.ts` is having errors? In that case we are not capturing those errors. I think that we should at least capture all the errors until HASS has a connection established.

Comment (bramkragten) [reply src/entrypoints/core.ts:114]: OK, let's move this to another PR :-)

**Pull request merged**